### DISCUSSION TITLE
Show Filament flow at temperature graph

### DISCUSSION THREAD

**Discussion by webkonto** [KlipperScreen/KlipperScreen#519] [Ideas]
Feature request: 
As a user, I want to see the energy used to melt the filament so that I can judge, if more speed is doable or if I am off limits.

Description:
The most problematic part in the discussion of "fest printing" is not really laying in the speed of the printhead but in the ability of the system, to meld the filament in the needed time. Since today, the "energy flow" is widely unknown to the user, I'd like to make is visible. E.g. if you are printing with a 0,6mm nozzle, layer hight 0,2mm and Speed 100mm, the "calculated" maximal flow is 0,6*0,2*100 mm^3 = 12mm^3 at temperature 235 Deg (my setting for PET-G). Now I can calculate the needed power to do this ( Q=p*cp*dT or Energie needed is Density * CP-Value of PET * (235 deg -Room Temp) and compare it to my setup. Assuming, that the efficiency of the nozzle heating is rather low, a rule of thumb could be 40% of the nominal power), you could get a feeling, how close to your limit you are printing.

Input values:
- Material properies (Density, CP-Value)
- Nozzle temperature
- Assuming room temperature is always something like 22 deg)
- actual Material flow (in the equation above, the Energy is given per Volumne unit)

Accepance Criteria:
- Show a graph or a value of the actual/min/max Energy flow for heating up the Filament

What do you think? Too absract? Too complex to understand? I think this is very helpful.

Future idea: If you know the Material of your heating block and the weight of it, you could calculate a "buffer" for peek energy usage... Fancy!

**Comment by alfrix** [comment]
this too complex for most users, the ability to show heater power is in the settings, that would give a general idea if the heater is too weak, i personally don't resort to calculations, i do prints at different flowrates and determine visually where i would like to stop, usually the walls acquire a bad look and then start to have gaps, you place the values in the slicer for the filament and never look at them again until something in the hotend gets changed
